content/en/integrations/apisix.md

```diff
@@ -0,0 +1,157 @@
+
+## Overview
+
+As the complexity of IT products and consumer-facing application development increases, monitoring becomes an integral part of any application delivery. Additionally, to meet the endless demand of rapid upgrade cycles while ensuring stability, streamlined performance and keeping a perfect balance between service level indicators (SLI) with Service-level objectives (SLO) and Service-level agreement (SLA) - effective monitoring is immensely important.
+
+As a Cloud API Management Product, [Apache APISIX][1] decouples observability concerns from the application, which gives the developers an advantage of building applications focusing just on the business logic while Apache APISIX will take care of observability for the platform of their choice.
+
+Apache APISIX recently released a new plugin: APISIX-Datadog, to provide a deeper integration with Datadog. This article introduces the APISIX-Datadog Plugin and its capabilities.
+
+## Run Datadog Agent
+
+1. If you are already using Datadog inside your infrastructure, you must have a datadog agent installed in your systems. It may either be a docker container, pod or binary for a respective package manager. In this case, you are good to go. Just make sure port 8125/udp is allowed through the firewall (if any) i.e more specifically, the Apache APISIX agent can reach port 8125 of the datadog agent. You may skip this subsection.
+
+To learn more about how to install a full-fledged datadog agent, visit [here][2].
+
+2. If you are new to Datadog
+
+  1. First create an account by visiting www.datadoghq.com.
+  2. Generate an API Key.
+  {{< img src="integrations/apisix/generate_api_key.png" alt="Generate an API Key" popup="true">}}
+
+3. APISIX-Datadog plugin requires only the dogstatsd component of `datadog/agent` as the plugin asynchronously send metrics to the dogstatsd server following the statsd protocol over standard UDP socket. That's why APISIX recommends using the standalone `datadog/dogstatsd` image instead of using the full agent. It's extremely lightweight (only ~11 MB in size) compared to ~2.8GB of `datadog/agent` image.
+
+To run it as a container:
+
+```shell
+# pull the latest image
+docker pull datadog/dogstatsd:latest
+# run a detached container
+docker run -d --name dogstatsd-agent -e DD_API_KEY=<Your API Key from step 2> -p 8125:8125/udp  datadog/dogstatsd
+```
+
+If you are using Kubernetes in your production environment, you can deploy `dogstatsd` as a `Daemonset` or as a `Multi-Container Pod` alongside Apache APISIX agent.
+
+## Activate APISIX-Datadog plugin
+
+The following is an example on how to activate the datadog plugin for a specific route. We are assumming your `dogstatsd` agent is already up an running.
+
+```shell
+# enable plugin for a specific route
+curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
```

Comment:
hi @yzeng25 - can you please confirm if this is a real API key? If so, please delete the API key as soon as possible. I am also initiating an internal process to investigate as well.

Reply:
hi @apigirl it's not a real key, just a mock key for APISIX, and it's nothing with Datadog.
@yzeng25 please add docs and comments for this.

Reply:
Sure @moonming

Reply:
okay! thank you for verifying